### PR TITLE
Adds human-readable ACH parsing to the webui

### PR DESCRIPTION
**Pull request by markalex**
Ports the human-readable formatting from the command line `achcli` to the `webui`. 

Note: It might be worth changing the input and output `textarea`s to vertical rather than horizontal to accommodate the wrapping that can happen with the human-readable content.

cc @adamdecaf

### CODE REVIEW

File: cmd/webui/ach/ach_js.go [hunk 1]
```diff
@@ -78,8 +87,31 @@ func printACH() js.Func {
 	})
 }
 
+func printReadable() js.Func {
+	return js.FuncOf(func(_ js.Value, args []js.Value) interface{} {
+		if len(args) != 1 {
+			return "Invalid no of arguments passed"
```
Comment (vxio) [inline cmd/webui/ach/ach_js.go]:
```suggestion
			return "Invalid number of arguments passed"
```
Could we update this in printACH as well?

File: cmd/webui/ach/ach_js.go [hunk 2]
```diff
@@ -78,8 +87,31 @@ func printACH() js.Func {
...
+		r := strings.NewReader(args[0].String())
+		file, err := ach.NewReader(r).Read()
+		if err != nil {
+			msg := fmt.Sprintf("unable to parse ach file: %v", err)
+			fmt.Println(msg)
+			return msg
+		}
+		parsed, err := parseReadable(&file)
+		if err != nil {
+			fmt.Printf("unable to convert ach file to readable format %s\n", err)
```
Comment (vxio) [inline cmd/webui/ach/ach_js.go]:
```suggestion
			fmt.Printf("unable to convert ach file to human-readable format %s\n", err)
```

**Comment by vxio** [comment]
Appreciate the PR @markalex; will review soon!

**Comment by vxio** [comment]
Strange, even on `master` branch, the web-ui buttons aren't working for me when I run it locally:image

File: cmd/webui/assets/index.html [hunk 3]
```diff
@@ -95,15 +95,18 @@
...
     </header>
-    <textarea id="jsoninput" name="jsoninput" cols="80" rows="40" placeholder="Paste your ACH file contents here..." required></textarea>
-    <textarea id="jsonoutput" name="jsonoutput" cols="80" rows="40" readonly></textarea>
+    <textarea id="contentinput" name="contentinput" cols="80" rows="40" placeholder="Paste your ACH file contents here..." required></textarea>
+    <textarea id="contentoutput" name="contentoutput" cols="80" rows="40" readonly></textarea>
     <div id="buttons">
-        <button type="submit" onclick="convertToJson(jsoninput.value)">Parse ACH file</button>
-        <button type="button" onclick="convertToAch(jsoninput.value)">Parse JSON contents</button>
+        <button type="submit" onclick="convertToJson(contentinput.value)">Parse ACH file to JSON</button>
+        <button type="button" onclick="convertToReadable(contentinput.value)">Parse ACH file to human-readable</button>
```
Comment (vxio) [inline cmd/webui/assets/index.html]:
```suggestion
        <button type="button" onclick="convertToReadable(contentinput.value)">Parse ACH file to human-readable format</button>
```

File: cmd/webui/ach/ach_js.go [hunk 4]
```diff
@@ -24,6 +25,14 @@ func parseACH(input string) (string, error) {
 	return buf.String(), nil
 }
 
+func parseReadable(file *ach.File) (string, error) {
+	var buf bytes.Buffer
+	// passing nil for the masking Opts to be
+	// consistent with the JSON output that is not masked
+	describe.File(&buf, file, nil)
```
Comment (vxio) [inline cmd/webui/ach/ach_js.go]: nit: I recommend being explicit with the Opts and I think we could remove the code comments here
```suggestion
	describe.File(&buf, file,  &describe.Opts{
	                         MaskNames: false,
				MaskAccountNumbers:  false,
			})
```

**Review by vxio — COMMENTED** [review]
This PR looks good, I just had a few minor comments

File: cmd/webui/ach/ach_js.go (hunk 1 above)
Comment (markalex) [reply cmd/webui/ach/ach_js.go]: I copied that from the other handlers - I'll change it in all places. I agree "number" is better.

**Comment by markalex** [comment]
I see another altered file locally: `wasm_exec.js`. I didn't make any changes to it so I did not include it in the PR. Looks like a generated file that is needed given the `cp` command as the first step in execution
```
> make build-webui && ./bin/webui
cp /usr/local/go/misc/wasm/wasm_exec.js ./cmd/webui/assets/wasm_exec.js
``` 
I'll add it to the PR.

**Comment by adamdecaf** [comment]
`wasm_exec.js` comes from the Go distribution you're using. The latest is used when building in CI but we haven't updated it in the git tree. Makes me think we should `.gitignore` it since that file is autogenerated.

File: cmd/webui/ach/ach_js.go (hunk 4 above)
Comment (adamdecaf) [reply cmd/webui/ach/ach_js.go]: Either or. That package is designed to handle `nil` without issue.

**Review by adamdecaf — DISMISSED** [review]

**Comment by adamdecaf** [comment]
I haven't tried it locally, but assume Mark and Vince have. Looks good!

File: cmd/webui/assets/index.html [hunk 5]
```diff
@@ -95,15 +95,18 @@
...
-    <textarea id="jsoninput" name="jsoninput" cols="80" rows="40" placeholder="Paste your ACH file contents here..." required></textarea>
-    <textarea id="jsonoutput" name="jsonoutput" cols="80" rows="40" readonly></textarea>
+    <textarea id="contentinput" name="contentinput" cols="80" rows="40" placeholder="Paste your ACH file contents here..." required></textarea>
+    <textarea id="contentoutput" name="contentoutput" cols="80" rows="40" readonly></textarea>
     <div id="buttons">
-        <button type="submit" onclick="convertToJson(jsoninput.value)">Parse ACH file</button>
-        <button type="button" onclick="convertToAch(jsoninput.value)">Parse JSON contents</button>
+        <button type="submit" onclick="convertToJson(contentinput.value)">Parse ACH file to JSON</button>
+        <button type="button" onclick="convertToReadable(contentinput.value)">Parse ACH file to human-readable format</button>
+        <button type="button" onclick="convertToAch(contentinput.value)">Parse JSON to ACH file</button>
```
Comment (adamdecaf) [inline cmd/webui/assets/index.html]: Maybe the following text is better?

- View as JSON
- Human Readable
- View as Nacha

Comment (markalex) [reply cmd/webui/assets/index.html]: I like that better.

Comment (vxio) [reply cmd/webui/assets/index.html]: I like shortening the text but could be ambiguous to unfamiliar users what the input should be, thoughts on?
- ACH to JSON
- ACH to human-readable
- JSON to ACH

Or 

- Plaintext to JSON
- Plaintext to human-readable
- JSON to plaintext

Comment (markalex) [reply cmd/webui/assets/index.html]: I had a similar thought. Going to try a few options and see which is the easiest to understand.

**Comment by markalex** [comment]
See what you think now. The button text is simpler and it no longer errors if user attempts to convert a format to itself (JSON -> JSON, for example).

**Comment by vxio** [comment]
Sweet!image

**Review by vxio — APPROVED** [review]

**Comment by vxio** [comment]
@markalex This looks great, going to merge it in, thank you again for the PR!

**Comment by adamdecaf** [comment]
Thanks all for getting this merged!

**Pull request merged**